packages/ws/src/ws/WebSocketShard.ts

```diff
@@ -567,7 +567,14 @@ export class WebSocketShard extends AsyncEventEmitter<WebSocketShardEventsMap> {
 
 			case GatewayOpcodes.Hello: {
 				this.emit(WebSocketShardEvents.Hello);
-				this.debug([`Starting to heartbeat every ${payload.d.heartbeat_interval}ms`]);
+				const jitter = Math.random();
+				const firstWait = Math.floor(payload.d.heartbeat_interval * jitter);
+				this.debug([`Preparing first heartbeat of the connection with a jitter of ${jitter}; waiting ${firstWait}ms`]);
+
```

Comment:
I think it's better that we do it this way so we can clear it if needed because we won't be able to clear the sleep if we need to
```ts
                                this.heartbeatInterval = setInterval(async () => {
					if (this.heartbeatInterval) clearInterval(this.heartbeatInterval);
					void this.heartbeat();
					this.heartbeatInterval = setInterval(() => void this.heartbeat(), payload.d.heartbeat_interval);
				}, payload.d.heartbeat_interval * Math.random());
```